sopel/cli/run.py

```diff
@@ -0,0 +1,658 @@
+            logger.setup_logging(p)
+            p.run(config.core.host, int(config.core.port))
+        except KeyboardInterrupt:
+            break
+        except Exception:  # TODO: Be specific
+            trace = traceback.format_exc()
+            try:
+                tools.stderr(trace)
+            except Exception:  # TODO: Be specific
+                pass
+            logfile = open(os.path.join(config.core.logdir, 'exceptions.log'), 'a')
+            logfile.write('Critical exception in core')
+            logfile.write(trace)
+            logfile.write('----------------------------------------\n\n')
+            logfile.close()
+            # TODO: This should be handled by command_start
+            # All we should need here is a return value, but replacing the
+            # os._exit() call below (at the end) broke ^C.
+            # This one is much harder to test, so until that one's sorted it
+            # isn't worth the risk of trying to remove this one.
+            os.unlink(pid_file)
+            os._exit(1)
+
+        if not isinstance(delay, int):
+            break
+        if p.wantsrestart:
+            return -1
+        if p.hasquit:
+            break
+        tools.stderr(
+            'Warning: Disconnected. Reconnecting in %s seconds...' % delay)
+        time.sleep(delay)
+    # TODO: This should be handled by command_start
+    # All we should need here is a return value, but making this
+    # a return makes Sopel hang on ^C after it says "Closed!"
+    os.unlink(pid_file)
+    os._exit(0)
+
+
+def add_legacy_options(parser):
+    parser.add_argument("-d", '--fork', action="store_true",
+                        dest="daemonize", help="Daemonize Sopel")
+    parser.add_argument("-q", '--quit', action="store_true", dest="quit",
+                        help=(
+                            "Gracefully quit Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop` instead)"))
+    parser.add_argument("-k", '--kill', action="store_true", dest="kill",
+                        help=(
+                            "Kill Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop --kill` instead)"))
+    parser.add_argument("-r", '--restart', action="store_true", dest="restart",
+                        help=(
+                            "Restart Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel restart` instead)"))
+    parser.add_argument("-l", '--list', action="store_true",
+                        dest="list_configs",
+                        help="List all config files found")
+    parser.add_argument("-m", '--migrate', action="store_true",
+                        dest="migrate_configs",
+                        help="Migrate config files to the new format")
```

Comment:
I was going to ask where the code to handle this option went, but… you had nothing to do with it.

Added in f85bf58 for (I guess) Willie 3.1, the `--migrate` option was neutered in 7e1dc1e (part of 4.0.0, I think). The code to handle the option was removed without mention in the commit log or changelog (that I can find).

Honestly, you can probably just drop this legacy option. It's likely nobody is using it any more after almost six years of Willie/Sopel doing nothing when passed the arg.

Or, keep it with a deprecation notice and remove it for real in Sopel 8, like its buddies. But it's probably fine to just remove (with a nice commit message about how I went Sopelunking and found the deleted code 😁).

Reply:
Nah, it's dead code, let's remove it.